SimG4Core/Application/src/Phase2SteppingAction.cc

```diff
@@ -201,27 +201,16 @@ void Phase2SteppingAction::UserSteppingAction(const G4Step* aStep) {
       if (!trkinfo->crossedBoundary()) {
         trkinfo->setCrossedBoundary(theTrack);
       }
-    } else if (preStep->GetPhysicalVolume() == calo && postStep->GetPhysicalVolume() != calo) {
-      bool backscattering(false);
-      if (postStep->GetPhysicalVolume() == tracker) {
-        backscattering = true;
-      } else if (postStep->GetPhysicalVolume() == cmse) {
-        // simple protection to avoid possible steps from calo towards the outer part of the detector, if allowed by geometry
-        // to be removed as soon as tracker-calo boundary becomes again the default
-        if (preStep->GetPosition().mag2() > postStep->GetPosition().mag2()) {
-          backscattering = true;
-        }
-      }
-      // store transition calo -> cmse to tag backscattering
-      if (backscattering) {
-        TrackInformation* trkinfo = static_cast<TrackInformation*>(theTrack->GetUserInformation());
-        if (!trkinfo->isInTrkFromBackscattering()) {
-          trkinfo->setInTrkFromBackscattering();
+    } else if ((preStep->GetPhysicalVolume() == calo && postStep->GetPhysicalVolume() == tracker) ||
+               (preStep->GetPhysicalVolume() == cmse && postStep->GetPhysicalVolume() == tracker)) {
+      // accounting for both geometries with direct CALO -> Tracker transitions or older versions with CMSE in the middle
+      TrackInformation* trkinfo = static_cast<TrackInformation*>(theTrack->GetUserInformation());
+      if (!trkinfo->isInTrkFromBackscattering()) {
```

Comment:
@fabiocos , do we really need this "if"? Why not to set the flag in any case?

Reply:
@civanch you are right, in case it will trigger the debug statement multiple times, but that is a minor case, and we may get rid of routine "if" calls. I will remove it